src/vue/common/TokenRequestDetails.vue

```diff
@@ -0,0 +1,335 @@
+          </td>
+          <td>
+            <template v-if="request.isTransferable">
+              {{ 'token-request-details.present-msg' | globalize }}
+            </template>
+            <template v-else>
+              {{ 'token-request-details.absent-msg' | globalize }}
+            </template>
+          </td>
+        </tr>
+        <tr>
+          <td>
+            {{ 'token-request-details.requires-kyc-title' | globalize }}
+          </td>
+          <td>
+            <template v-if="request.isRequiresKYC">
+              {{ 'token-request-details.present-msg' | globalize }}
+            </template>
+            <template v-else>
+              {{ 'token-request-details.absent-msg' | globalize }}
+            </template>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+    <div class="token-request-details__buttons">
+      <button
+        v-ripple
+        class="token-request-details__update-btn"
+        :disabled="!canBeUpdated"
+        @click="$emit(EVENTS.update)"
+      >
+        {{ 'token-request-details.update-btn' | globalize }}
+      </button>
+      <button
+        v-ripple
+        class="token-request-details__cancel-btn"
+        :class="{
+          'token-request-details__cancel-btn--disabled': !canBeCanceled
+        }"
+        :disabled="!canBeCanceled"
+        @click="cancelRequest"
+      >
+        {{ 'token-request-details.cancel-btn' | globalize }}
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import TokenLogo from '@/vue/common/TokenLogo'
+
+import { base, ASSET_POLICIES, REQUEST_TYPES } from '@tokend/js-sdk'
+
+import { REQUEST_STATES } from '@/js/const/request-states.const'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { Sdk } from '@/sdk'
+import config from '@/config'
+
+import { AssetCreateRequestRecord } from '@/js/records/requests/asset-create.record'
+
+const EVENTS = {
+  update: 'update',
+}
+
+export default {
+  name: 'token-request-details',
+  components: {
+    TokenLogo,
+  },
+  props: {
+    request: { type: Object, required: true },
+  },
+  data: _ => ({
+    isCanceled: false,
+    ASSET_POLICIES,
+    EVENTS,
+    REQUEST_STATES,
+    REQUEST_TYPES,
+  }),
+  computed: {
+    isAssetCreationType () {
+      return this.request instanceof AssetCreateRequestRecord
+    },
+    tokenTermsUrl () {
+      return this.request.termsUrl(config.FILE_STORAGE)
+    },
+    canBeUpdated () {
+      return this.request.isPending || this.request.isRejected
+    },
+    canBeCanceled () {
+      return this.request.isPending && !this.isCanceled
+    },
+  },
+  methods: {
+    async cancelRequest () {
+      this.isCanceled = true
+      try {
+        const operation = base.ManageAssetBuilder.cancelAssetRequest({
+          requestID: this.request.id,
+        })
+        await Sdk.horizon.transactions.submitOperations(operation)
+        this.request = Object.assign(this.request, {
```

Comment:
it is not recommended to perform any changes to the existing model outside its parsing/assigning etc. if it has no appropriate method. When we fix it we will not be needed to have `isCanceled` boolean since the request will have the state set by the "legitimate" way so we can read it just like we read `isPending` property of it a bit earlier.

To fix it we have two ways:
1. We can emit an event to the parent (as he provides as the data by prop). In this way we probably may have to move the canceling request to the parent to avoid passing raw data with within the event payload.
2. We can parse it by itself, but in this way we need to add a proxy-variable because we cannot modify prop. This way is much more difficult and complicated so I’d pref the first one.

Reply:
Updated